Client/Frontend/Home/ActivityStreamSupplementary.swift

```diff
@@ -0,0 +1,310 @@
+        var squareFrame = CGRectMake(0, 0, frame.height, frame.height)
+        squareFrame.center = frame.center
+        contentView.frame = squareFrame
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        contentView.layer.cornerRadius = TopSiteCellUX.CellCornerRadius
+        contentView.layer.masksToBounds = true
+
+        titleLabel = UILabel()
+        titleLabel.layer.masksToBounds = true
+        titleLabel.textAlignment = .Center
+        titleLabel.font = TopSiteCellUX.TitleFont
+        titleLabel.textColor = UIColor.blackColor()
+        titleLabel.backgroundColor = TopSiteCellUX.TitleBackgroundColor
+        contentView.addSubview(titleLabel)
+
+        let heightInset = Int(frame.height * TopSiteCellUX.TitleInsetPercent)
+        titleLabel.snp_makeConstraints { make in
+            make.edges.equalTo(self).inset(UIEdgeInsetsMake(CGFloat(heightInset), 0, 0, 0))
+        }
+
+        imageView = UIImageView()
+        imageView.layer.masksToBounds = true
+        contentView.addSubview(imageView)
+        imageView.snp_makeConstraints { make in
+            make.height.equalTo(self.frame.height/2)
+            make.width.equalTo(self.frame.height/2)
+
+            // Add an offset to the image to make it appear centered with the titleLabel
+            let offset = Int(self.frame.height) - heightInset
+            make.centerX.equalTo(self.snp_centerX)
+            make.centerY.equalTo(self.snp_centerY).offset(CGFloat(-offset/2))
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        backgroundColor = UIColor.whiteColor()
+        imageView.image = nil
+        titleLabel.text = ""
+    }
+
+    func setImageWithURL(url: NSURL) {
+        imageView.sd_setImageWithURL(url) { (img, err, type, url) -> Void in
+            guard let img = img else {
+                // No favicon found. Do something!
+                return
+            }
+
+            img.getColors(CGSize(width: 50, height:50)) { colors in
+                //In cases where the background is white. Force the background color to a different color
+                var bgColor: UIColor
+                if colors.backgroundColor.isWhite {
+                    let colorArr = [colors.detailColor, colors.primaryColor].filter {return !$0.isWhite}
+                    if colorArr.isEmpty {
+                        ///need an array of default colors
+                        bgColor = UIColor.greenColor()
+                    }
+                    else {
+                        bgColor = colorArr[0]
+                    }
+                }
+                else {
+                    bgColor = colors.backgroundColor
+                }
+                contentView.backgroundColor = bgColor
+
+            }
+        }
+    }
+
+    func configureWithTopSiteItem(site: TopSiteItem) {
```

Comment:
I've moved configuration of the cell into its own function. What do you think of this approach compared to customizing the cell in cellForRowAtIndexPath?